### PR TITLE
Encrypt migration credentials at rest

### PR DESCRIPTION
**Pull request by noerw**
Storing these credentials is a liability (or plain stupid)

- encrypt credentials with SECRET_KEY before persisting to task queue table (they need to be persisted due to the nature of the task queue)
   - security in depth: helps when attacker has access to DB only, but not `app.ini`
- delete all credentials (even encrypted) from the `task` table, once the migration is done, for safety
    - security in depth: minimizes leaked data if attacker gains access to snapshot of both DB and app.ini
- delete all credentials from past non-pending migrations from the db via a migration (the other kind of migration ;)

Will backport this to 1.14, implementing the migration as a doctor task.

### CODE REVIEW

File: modules/migrations/base/options.go [hunk 1]
```diff
@@ -11,10 +11,13 @@ import "code.gitea.io/gitea/modules/structs"
...
-	AuthUsername string `json:"auth_username"`
-	AuthPassword string `json:"auth_password"`
-	AuthToken    string `json:"auth_token"`
+	CloneAddr             string `json:"clone_addr" binding:"Required"`
+	CloneAddrEncrypted    string `json:"clone_addr_encrypted"`
+	AuthUsername          string `json:"auth_username"`
+	AuthPassword          string `json:"auth_password"`
+	AuthPasswordEncrypted string `json:"auth_password_encrypted"`
+	AuthToken             string `json:"auth_token"`
+	AuthTokenEncrypted    string `json:"auth_token_encrypted"`
```
Comment (zeripath) [inline modules/migrations/base/options.go]: probably need some omitempty here.

**Comment by lunny** [comment]
You could add a doctor task and back port the doctor sub command.

**Review by zeripath — APPROVED** [review]

File: models/migrations/v180.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	"xorm.io/builder"
+	"xorm.io/xorm"
+
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/modules/migrations/base"
+	"code.gitea.io/gitea/modules/structs"
+	"code.gitea.io/gitea/modules/util"
+)
+
+func deleteMigrationCredentials(x *xorm.Engine) (err error) {
```
Comment (lunny) [inline models/migrations/v180.go:18]: How about encrypt them but not delete them?

Comment (noerw) [reply models/migrations/v180.go:18]: why?
I outlined the rationale for deleting in the PR description:
> delete all credentials (even encrypted) from the task table, once the migration is done, for safety
>   - security in depth: minimizes leaked data if attacker gains access to snapshot of both DB and app.ini

File: models/migrations/v180.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	jsoniter "github.com/json-iterator/go"
+	"xorm.io/builder"
+	"xorm.io/xorm"
+
+	"code.gitea.io/gitea/models"
```
Comment (lunny) [inline models/migrations/v180.go:8]: import sequence is not correct.

**Review by lunny — APPROVED** [review]

**Pull request merged**